libbindgen/src/ir/context.rs

```diff
@@ -1066,3 +1102,38 @@ impl<'ctx, 'gen> Iterator for WhitelistedItemsIter<'ctx, 'gen>
         Some(id)
     }
 }
+
+pub struct RelatedItemsIter<'ctx, 'gen>
+    where 'gen: 'ctx,
+{
+    ctx: &'ctx BindgenContext<'gen>,
+    seen: ItemSet,
```

Comment:
This should be a `BTreeMap<ItemId, ItemId>` mapping from a seen item, to the item from which we first discovered it (or mapping to itself if we found it as a root).

That way, if/when we find a dangling `ItemId` reference, we can walk backwards through the path with this map, something like this:

```rust
if is_dangling(id) {
    let mut path = vec![];
    let mut current = id;
    loop {
        let predecessor = self.seen.get(current)
            .expect("We know we found this item id, so it must have a predecessor");
        if predecessor == current {
            break;
        }
        path.push(predecessor);
        current = predecessor;
    }
    path.reverse();
    panic!("Found reference to dangling id = {:?}\n
            via path = {:?}",
           id,
           path);
}
```